File: app/src/main/java/com/schrondinger/quin/ui/main/fragment/TypeOneFragment.java
package com.schrondinger.quin.ui.main.fragment;

import android.content.Context;
import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import com.schrondinger.quin.R;
import com.schrondinger.quin.base.activity.ActivityInject;
import com.schrondinger.quin.base.activity.BaseMVPFragment;
import com.schrondinger.quin.bean.User;
import com.schrondinger.quin.mvp.constract.TypeOneFragmentConstract;
import com.schrondinger.quin.mvp.model.TypeOneFragmentModel;
import com.schrondinger.quin.mvp.presenter.TypeOneFragmentPresenter;

import java.util.HashMap;
import java.util.Map;

import butterknife.BindView;

@ActivityInject(rootViewId = R.layout.fragment_type_one)
public class TypeOneFragment extends BaseMVPFragment<TypeOneFragmentPresenter,TypeOneFragmentModel> implements TypeOneFragmentConstract.View  {
    @BindView(R.id.testText)
    TextView testText;
    @BindView(R.id.testButton)
    Button testButton;

    public static TypeOneFragment newInstance() {

        TypeOneFragment fragment = new TypeOneFragment();
        return fragment;
    }

    @Override
    public void initData() {
        super.initData();
    }

    @Override
    public void initView() {
        super.initView();
    }

    @Override
    public void initListener() {
        super.initListener();
        testButton.setOnClickListener(this);
    }

    @Override
    public void initLoad() {
        super.initLoad();

    }

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.testButton:
                Map<String,String> sendMap = new HashMap<>();
                sendMap.put("id","1");
                mPresenter.getUserInfo(sendMap);
                break;
        }
    }

    @Override
    public void updateUI(User user) {
        testText.setText(user.getUserName());
    }
}

File: app/src/main/java/com/schrondinger/quin/ui/main/fragment/TypeOneFragment.kt
package com.schrondinger.quin.ui.main.fragment

import android.support.v4.app.Fragment
import android.support.v4.content.ContextCompat
import android.view.View
import com.schrondinger.quin.R
import com.schrondinger.quin.base.activity.ActivityInject
import com.schrondinger.quin.base.activity.BaseFragment
import com.schrondinger.quin.bean.common.CommMap
import com.schrondinger.quin.ui.main.MainActivity
import com.schrondinger.quin.ui.common.adaper.ContentFragmentPagerAdapter
import com.schrondinger.quin.ui.mine.fragment.MineOneFragment
import kotlinx.android.synthetic.main.fragment_type_one.*

@ActivityInject(rootViewId = R.layout.fragment_type_one)
class TypeOneFragment : BaseFragment() {
    private lateinit var tabList:ArrayList<CommMap>
    private var contentFragmentPagerAdapter: ContentFragmentPagerAdapter? = null
    private lateinit var tabFragments: Array<Fragment?>

    override fun initData() {
        super.initData()
        tabList = arrayListOf(CommMap("直播", MineOneFragment::class.java.name),
                CommMap("推荐", MineOneFragment::class.java.name),
                CommMap("追番", MineOneFragment::class.java.name))

        tabFragments = kotlin.arrayOfNulls(tabList.size)

        for (i in tabList.indices){
            val clazz = Class.forName(tabList[i].value) as Class<BaseFragment>
            tabFragments[i] = clazz.newInstance()
        }
    }

    override fun initView() {
        super.initView()
        initViewPager()
        initTab()
    }

    override fun initListener() {
        super.initListener()
        di_open_drawer.setOnClickListener(this)
        ci_user_head.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        when(v.id){
            R.id.di_open_drawer,R.id.ci_user_head->{
                (activity as MainActivity).openDrawer()
            }
        }
    }

    private fun initViewPager(){
        contentFragmentPagerAdapter = ContentFragmentPagerAdapter(childFragmentManager, tabList)
        vp_content.adapter = contentFragmentPagerAdapter
        vp_content.currentItem = 0
        vp_content.offscreenPageLimit = tabFragments.size
    }

    private fun initTab() {
        tl_tab.setupWithViewPager(vp_content)
        tl_tab.setTabTextColors(ContextCompat.getColor(_mActivity, R.color.tab_text_color), ContextCompat.getColor(_mActivity, R.color.colorPrimary))
        tl_tab.setSelectedTabIndicatorColor(ContextCompat.getColor(_mActivity,R.color.colorPrimary))
    }
}
